### ISSUE TITLE
exec_prefix disappears from .pc files when porting Autotools libraries to Meson

### ISSUE THREAD

**Issue by smcv** [mesonbuild/meson#4251]
When a library (for example GLib) is built with Autotools, it will typically have `exec_prefix=${prefix}` (or some non-default value if configured with `--exec-prefix`) in its `.pc` file.

If that library is converted to Meson, because Meson doesn't (yet?) support a separate `exec_prefix` (see also #364, #1398), it will typically lose that line from its `.pc` file. This could conceivably break consumers of that library if they use something like

```
pkg-config --variable=exec_prefix glib-2.0
```

to locate the library.

(Many dependent packages that do this, although not for GLib, can be found here: <https://codesearch.debian.net/search?q=--variable%5B+%3D%5Dexec_prefix&perpkg=1>)

If #1398 is not going to be implemented soon, then it might be a good backwards compatibility step to generate

```
exec_prefix=${prefix}
```

in all `.pc` files, to tell library users that the `${exec_prefix}` is (always) the same as the `${prefix}`.

**Comment by smcv** [comment]
GLib issue: https://gitlab.gnome.org/GNOME/glib/issues/1540

**Comment by jpakkane** [comment]
Based on my reading of the bug, nobody is actually using `exec_prefix` for real. Based on that I'm not inclined to add support for it without an actual real world use case such as breaking dependencies.

**Comment by smcv** [comment]
Nobody is using *GLib's* exec_prefix in practice, but the Debian code search in my initial comment finds lots of packages that are using their dependencies' exec_prefixes, which would break if their dependencies switched to Meson.

(Admittedly, lots of those examples seem to be blindly assuming `${bindir}` is always `${exec_prefix}/bin`, for some reason...)

**Comment by smcv** [comment]
No, exec_prefix is not some cargo-culted thing. It's the prefix for architecture-dependent paths like bindir and libdir, and appears in those variables' defaults for Autotools. The .pc file for every Autotools project *has to* include exec_prefix, because otherwise expanding ${libdir} would fail, and `pkg-config --libs` wouldn't work.

In practice exec_prefix is normally just ${prefix}, and I thought prefix != exec_prefix was from the era of NFS-mounting root filesystems between proprietary Unix servers, but according to #1398 Exherbo still uses prefix != exec_prefix.

Current Meson doesn't support a separate exec_prefix, so in current Meson projects the default bindir and libdir start with ${prefix} instead of ${exec_prefix} (removing the indirection); but if people are making use of exec_prefix, then that makes porting a library to Meson a potential API break.

For instance, if Qt switched to Meson, vlc and gst-plugins-bad wouldn't be able to find its moc, rcc, uic tools in a non-standard location: https://sources.debian.org/src/vlc/3.0.4-2/configure.ac/?hl=3694#L3694 https://sources.debian.org/src/gst-plugins-bad1.0/1.8.3-1/configure.ac/?hl=2834#L2834

**Comment by jpakkane** [comment]
The [pkg-config module](https://mesonbuild.com/Pkgconfig-module.html)'s generate function has the keyword `variables`, which allows you to add arbitrary key-value pairs in the pc file. Since this issue only matters for very few projects, could add `exec_prefix` manually or is there some deeper magic that needs to happen in the pc file?
